src/vmm/src/io_uring/operation/cqe.rs

```diff
@@ -13,22 +13,13 @@ unsafe impl ByteValued for io_uring_cqe {}
 #[derive(Debug)]
 pub struct Cqe<T> {
     res: i32,
-    user_data: Box<T>,
+    user_data: T,
 }
 
 impl<T: Debug> Cqe<T> {
-    /// Construct a [`Cqe`] object from a raw `io_uring_cqe`.
-    ///
-    /// # Safety
-    /// Unsafe because we assume full ownership of the inner.user_data address.
-    /// We assume that it points to a valid address created with a `Box<T>`,
-    /// with the correct type `T`, and that ownership of that address is passed
-    /// to this function.
-    pub(crate) unsafe fn new(inner: io_uring_cqe) -> Self {
-        Self {
-            res: inner.res,
-            user_data: Box::from_raw(inner.user_data as *mut T),
-        }
+    /// Construct a Cqe object from a raw `io_uring_cqe`.
+    pub(crate) fn new(res: i32, user_data: T) -> Self {
+        Self { res, user_data }
     }
```

Comment:
The comment is outdated. Also would be worth removing `(crate)` as all other methods are just `pub`.